### DISCUSSION TITLE
Add custom SQL suggestions

### DISCUSSION THREAD

**Discussion by bubbajoe** [microsoft/monaco-editor#3876] [Q&A] [answered]
Hello,

I would like to add custom intelligence or suggestions. For tables and columns, supported functions in the database, etc...

The documentation I saw didn't seem to how to do this.

**Comment by tamayika** [comment, marked as answer]
Plese see [example](https://microsoft.github.io/monaco-editor/playground.html?source=v0.37.1#example-extending-language-services-completion-provider-example)

**Reply by bubbajoe** [reply]
Hey thanks for your reply. This example is for Javascript, not SQL.

Is there no documentation for the functions/constants in the example?

**Reply by tamayika** [reply]
No, this is a example for JSON completion.
You can use the same function `monaco.languages.registerCompletionItemProvider()` in the example for what language including SQL.
https://microsoft.github.io/monaco-editor/docs.html#functions/languages.registerCompletionItemProvider.html
